sfge/src/main/java/com/salesforce/rules/fls/apex/operations/SchemaBasedValidationAnalyzer.java

```diff
@@ -84,20 +95,35 @@ public Set<FlsValidationRepresentation.Info> checkForValidation(
                             + vertex);
         }
 
+        List<ApexBooleanValue> apexValues = new ArrayList<>();
+
         // Handle cases such as
         // if (myObject) { /*Do Something*/ }
+
+        if (apexValue instanceof ApexBooleanValue) {
+            apexValues.add((ApexBooleanValue) apexValue);
+        }
         // Convert the value to a boolean that represents if the object was initialized
         if (apexValue instanceof ApexCustomValue) {
-            apexValue = ApexValueBuilder.getWithoutSymbolProvider().buildBoolean();
+            apexValues.add(ApexValueBuilder.getWithoutSymbolProvider().buildBoolean());
         }
 
-        // Convert the value to a boolean that represents if the object was initialized
-        // TODO: This might need to be more specific
-        if (apexValue instanceof ApexForLoopValue || apexValue instanceof ApexSingleValue) {
-            apexValue = ApexValueBuilder.getWithoutSymbolProvider().buildBoolean();
+        if (apexValue instanceof ApexForLoopValue) {
```

Comment:
Handling for-loop values better while created existing validations from CRUD/FLS rule.